### PR TITLE
Add more jobs to internal CI

### PR DESCRIPTION
**Pull request by matt-kwong**
Add Windows basic tests and Linux portability build only to internal CI. Also, update copyright year and remove `#!/bin/bash` from .cfg header.

### CODE REVIEW

File: tools/internal_ci/windows/grpc_master.bat [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+@rem OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+setlocal
+
+@rem enter repo root
+cd /d %~dp0\..\..\..
+
+git submodule update --init
+
+sh tools\run_tests\helper_scripts\run_tests_in_workspace.sh -t -j 4 -x c_windows_dbg_sponge_log.xml --report_suite_name c_windows_dbg -l c -c dbg
```
Comment (jtattermusch) [inline tools/internal_ci/windows/grpc_master.bat]: why not just use run_tests_matrix.py here?

File: tools/internal_ci/linux/grpc_portability_build_only.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+set -ex
+
+# change to grpc repo root
+cd $(dirname $0)/../../..
+
+git submodule update --init
+
+tools/jenkins/run_jenkins_matrix.sh -f portability linux --build_only || FAILED="true"
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_portability_build_only.sh]: remove `|| FAILED="true"`

File: tools/internal_ci/windows/grpc_master.bat (hunk 1 above)
Comment (matt-kwong) [inline tools/internal_ci/windows/grpc_master.bat]: The XML reports need to end in sponge_log.xml. I'll add a flag for run_tests_matrix.py to support custom xml report names.

**Comment by matt-kwong** [comment]
Issues:
Basic Mac - #9804
Interop - #9825 #9797 
Perf - #9668

**Pull request merged**